### PR TITLE
Add disabled section in AddonsManagerAdapter

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt [hunk 1]
```diff
@@ -265,3 +269,8 @@ class AddonsManagerAdapter(
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal inner class NotYetSupportedSection(@StringRes val title: Int)
 }
+
+private fun Addon.isUnsupportedSection() = isInstalled() && !isSupported()
+private fun Addon.isRecommendedSection() = !isInstalled()
+private fun Addon.isInstalledSection() = isInstalled() && isSupported() && isEnabled()
+private fun Addon.isDisabledSection() = isInstalled() && isSupported() && !isEnabled()
```
Comment (csadilek) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt]: 👍

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt [hunk 2]
```diff
@@ -244,6 +243,11 @@ class AddonsManagerAdapter(
             itemsWithSections.add(Section(R.string.mozac_feature_addons_installed_section))
             itemsWithSections.addAll(installedAddons)
         }
```
Comment (csadilek) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt]: nit: newline.

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt [hunk 3]
```diff
@@ -265,3 +269,8 @@ class AddonsManagerAdapter(
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal inner class NotYetSupportedSection(@StringRes val title: Int)
 }
+
+private fun Addon.isUnsupportedSection() = isInstalled() && !isSupported()
+private fun Addon.isRecommendedSection() = !isInstalled()
```
[the remark this answers is not in the document]
Comment (csadilek) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt]: Oh, actually, I misread, this is fine.

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt [hunk 4]
```diff
@@ -222,20 +222,19 @@ class AddonsManagerAdapter(
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    @Suppress("ComplexMethod")
     internal fun createListWithSections(addons: List<Addon>): List<Any> {
         val itemsWithSections = ArrayList<Any>()
         val installedAddons = ArrayList<Addon>()
         val recommendedAddons = ArrayList<Addon>()
+        val disableAddons = ArrayList<Addon>()
```
Comment (gabrielluong) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt]: s/disableAddons/disabledAddons

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt [hunk 5]
```diff
@@ -265,3 +269,8 @@ class AddonsManagerAdapter(
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal inner class NotYetSupportedSection(@StringRes val title: Int)
 }
+
+private fun Addon.isUnsupportedSection() = isInstalled() && !isSupported()
+private fun Addon.isRecommendedSection() = !isInstalled()
+private fun Addon.isInstalledSection() = isInstalled() && isSupported() && isEnabled()
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt]: 👍

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt (hunk 2 above)
Comment (Amejia481) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt]: 👍

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt (hunk 4 above)
Comment (Amejia481) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt]: 👍

**Comment by Amejia481** [comment]
bors r=csadilek,gabrielluong

**Pull request merged**